components/TARGET_PSA/services/psa_prot_internal_storage/psa_prot_internal_storage.h

```diff
@@ -52,13 +60,11 @@ typedef uint32_t psa_its_status_t;
 #define PSA_ITS_ERROR_WRITE_ONCE                1   /**<    The operation failed because the provided key value was already created with PSA_ITS_WRITE_ONCE_FLAG */
 #define PSA_ITS_ERROR_FLAGS_NOT_SUPPORTED       2   /**<    The operation failed because one or more of the flags provided in `create_flags` is not supported or is not valid */
 #define PSA_ITS_ERROR_INSUFFICIENT_SPACE        3   /**<    The operation failed because there was insufficient space on the storage medium */
-#define PSA_ITS_ERROR_INVALID_KEY               4   /**<    The operation failed because the key value provided was invalid */
-#define PSA_ITS_ERROR_STORAGE_FAILURE           5   /**<    The operation failed because the physical storage has failed (Fatal error) */
-#define PSA_ITS_ERROR_BAD_POINTER               6   /**<    The operation failed because one of the provided pointers is invalid, for example is `NULL` or references memory the caller cannot access */
-#define PSA_ITS_ERROR_KEY_NOT_FOUND             7   /**<    The operation failed because the provided key value was not found in the storage */
-#define PSA_ITS_ERROR_INCORRECT_SIZE            8   /**<    The operation failed because the data associated with provided key is not the same size as `data_size` */
-#define PSA_ITS_ERROR_OFFSET_INVALID            9   /**<    The operation failed because an offset was supplied that is invalid for the existing data associated with the uid. For example, offset + size is
-                                                            past the end of the data */
+#define PSA_ITS_ERROR_STORAGE_FAILURE           4   /**<    The operation failed because the physical storage has failed (Fatal error) */
+#define PSA_ITS_ERROR_BAD_POINTER               5   /**<    The operation failed because one of the provided pointers is invalid, for example is `NULL` or references memory the caller cannot access */
+#define PSA_ITS_ERROR_KEY_NOT_FOUND             6   /**<    The operation failed because the provided key value was not found in the storage */
+#define PSA_ITS_ERROR_INCORRECT_SIZE            7   /**<    The operation failed because the data associated with provided key is not the same size as `data_size`, or `offset+data_size` is too large for the data, but `offset` is less than the size */
+#define PSA_ITS_ERROR_OFFSET_INVALID            8   /**<    The operation failed because an offset was supplied that is invalid for the existing data associated with the uid. For example, offset  is greater that the size of the data */
```

Comment:
Gonna say here what I said in the PSA API repo: Some of the return codes here make the implementation inefficient. For instance, `PSA_ITS_ERROR_OFFSET_INVALID` imposes a redundant `get_info` call before `get`. I think that if we already have a breaking change, then we should revise the entire error codes list.

Reply:
CC @dreemkiller

@davidsaada change requests better be sent via our spec repo.
This PR implements aggregated spec changes till now.

Reply:
Commented in the spec repo as well (to the PR opened by @orenc17). My comment's aim here is to stress that a review to this PR won't be full until we sort out the changes to the error code list.

Reply:
Can you please post a URL to your issue here?
It will allow us to keep track of it.
Thanks

Reply:
It was a [general comment](https://github.com/ARMmbed/psa_trusted_storage_api/pull/22/files#r243814438) in the [PR #22](https://github.com/ARMmbed/psa_trusted_storage_api/pull/22) of the spec repo. I didn't elaborate regarding the problems, but they are mainly in the [get implementation](https://github.com/ARMmbed/mbed-os/blob/70956ee9bbbc57c140bd6ebd9bf3f76384b4692f/components/TARGET_PSA/services/psa_prot_internal_storage/COMPONENT_PSA_SRV_IMPL/pits_impl.cpp#L178). I think that `PSA_ITS_ERROR_OFFSET_INVALID` return code is not required and that `PSA_ITS_ERROR_INCORRECT_SIZE` is already tested by KVStore, so no need to make the checks again.